fpga_interchange/arch.cc

```diff
@@ -573,4 +693,95 @@ const std::vector<std::string> Arch::availablePlacers = {"sa",
 const std::string Arch::defaultRouter = "router2";
 const std::vector<std::string> Arch::availableRouters = {"router1", "router2"};
 
+void Arch::map_cell_pins(CellInfo *cell, int32_t mapping) const
+{
+    cell->cell_mapping = mapping;
+    cell->pins.clear();
+    cell->cell_bel_pins.clear();
+
+    const CellBelMapPOD &cell_pin_map = chip_info->cell_map->cell_bel_map[mapping];
+
+    log_info("Mapping cell %s\n", cell->name.c_str(this));
+    for (const auto &pin_map : cell_pin_map.common_pins) {
+        IdString cell_pin(pin_map.cell_pin);
+        IdString bel_pin(pin_map.bel_pin);
+
+        if(cell_pin.str(this) == "GND") {
```

Comment:
You might want to think about using `constid`s like other arches (mixing fixed constids and db ones like nexus and nextpnr-xilinx) rather than string compares; this _probably_ isn't a very hot path but I suspect you will hit other paths where fast IdString compares/lookups are needed